addons/product/views/product_views.xml

```diff
@@ -90,7 +90,7 @@
                                             options="{'currency_field': 'currency_id', 'field_digits': True}"/>
                                             <span name="uom_span" groups="uom.group_uom">per
                                                 <field name="uom_id"
-                                                    class="oe_inline"
+                                                    class="oe_inline" style="max-width:136px"
```

Comment:
136 px because we don't need that much space (in list view we limit to 92).
136 as it is 100 + 36 for the button of the Many2One, 100 being the size of other monetary fields